python/composio_swe/benchmark/run_evaluation.py

```diff
@@ -63,58 +63,99 @@ def build_issue_description(hints, problem_statement):
     return tmpl
 
 
+def setup_workspace(repo, repo_to_workspace_map, repo_to_image_id_map, base_commit):
+    composio_client = Composio()
+    try:
+        if repo in repo_to_workspace_map:
+            print("Resetting repository to base commit")
+            workspace_id = repo_to_workspace_map[repo]
+            composio_client.actions.execute(
+                action=Action.CMDMANAGERTOOL_GITHUBCLONECMD,
+                params={
+                    "workspace_id": workspace_id,
+                    "repo_name": repo,
+                    "just_reset": True,
+                    "commit_id": base_commit,
+                },
+            )
+        elif repo in repo_to_image_id_map:
+            print("Using saved image")
+            start_time = datetime.datetime.now()
+            workspace_create_resp = CreateWorkspaceResponse.model_validate(
+                composio_client.actions.execute(
+                    action=Action.LOCALWORKSPACE_CREATEWORKSPACEACTION,
+                    params={"image_name": repo_to_image_id_map[repo]},
+                )
+            )
+            workspace_id = workspace_create_resp.workspace_id
+            workspace_creation_time = datetime.datetime.now() - start_time
+            print(
+                "workspace is created, workspace-id is: %s, creation time: %s",
+                workspace_id,
+                workspace_creation_time,
+            )
+            print("Resetting repository to base commit")
+            composio_client.actions.execute(
+                action=Action.CMDMANAGERTOOL_GITHUBCLONECMD,
+                params={
+                    "workspace_id": workspace_id,
+                    "repo_name": repo,
+                    "just_reset": True,
+                    "commit_id": base_commit,
+                },
+            )
+        else:
+            raise Exception("Repository not found in workspace or image maps")
+    except Exception as e:
+        print(f"Error occurred: {e}. Falling back to creating new workspace.")
+        start_time = datetime.datetime.now()
```

Comment:
could you add a todo on me, ideally nothing should happen in except clause. I will add a retry decorator here